OneCommunityGlobal/HighestGoodNetworkApp
# Yiyun cherrypick ek task edit suggest 3

This PR is the third part of [Eiki's Task Edit Suggestions PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/497/commits), due to the large scale of changes and commits, we decided to cherrypick all his changes and split into couple new PRs, which would be more readable and understandable. You can also see the first cherrypick PR [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/536) and second PR [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/552)

**Mainly changes explained:** 🔮 
1. update the task permissions for people, make mentor/manager can suggest and core team/volunteer can only view
2. create the taskSuggest notification button for owner/admin
3. implement the count notification logic
4. run Prettier on `src/utils/permissions.js` and add Optional chaining (?.) for cannot login as a volunteer

**How to test:** 🔍 

1. check on this PR as frontend
2. log as mentor/manger account
3. Dashboard -> Tasks and Timelogs -> Tasks tab -> click one of the tasks -> click the "suggest" button -> make some changes on the popup table -> **click "update" button**
<img width="942" alt="image" src="https://user-images.githubusercontent.com/5071040/200686673-7c3b3482-4bcd-4b20-85a9-cae2f11b6eb1.png">

4. log out and log in as a admin/owner user
5. click the red number to see all taskEditSuggestions (here might have a potentially bug, the number is zero, I'll file and fix later)
<img width="947" alt="image" src="https://user-images.githubusercontent.com/5071040/204936402-262ea957-3a3b-4175-a037-f2b23b013597.png">


**Potentially bugs** ❓ 
1. The count of taskEditSuggestions sometimes shows zero, you would need to click it to get the correct number there. Like I said above, once this PR get approved, I'll file this as a bug and fix later.
2. When you check this branch on your local, you might find you cannot access the userProfile page. Since teammembers had already [reported](https://highest-good.slack.com/archives/CSCN20FA5/p1669487586958809) this problem on Slack, it's not a bug introduced by this PR and I don't why the DEV works fine but local cannot work. As usually, I'll keep on eye on this problem.
<img width="961" alt="image" src="https://user-images.githubusercontent.com/5071040/204936456-69c36207-4fd8-40b0-88b3-5f939fa5f58e.png">
